src/database.ts

```diff
@@ -160,214 +149,94 @@ export class Database {
    * with the SQL statement, a prepared statement otherwise.
    */
-  execute(strings: TemplateStringsArray, ...args: BindValue[]): void;
-  execute(sql: string, ...args: BindValue[]): void;
-  execute(sql: string, args: Record<string, BindValue>): void;
-  execute(
-    sql: string | TemplateStringsArray,
-    ...args: BindValue[] | [Record<string, BindValue>]
-  ): void {
-    const sqlStr = typeof sql === "string" ? sql : sql.join("?");
-    if (args.length) {
-      const stmt = this.prepare(sqlStr);
-      if (isObject(args[0])) {
-        stmt.bindAllNamed(args[0] as Record<string, BindValue>);
-      } else {
-        stmt.bindAll(...args as BindValue[]);
+  exec(sql: string, ...params: RestBindParameters): void {
+    if (params.length === 0) {
+      const pErr = new Uint32Array(2);
+      sqlite3_exec(this.#handle, toCString(sql), 0, 0, pErr);
+      const errPtr = pErr[0] + 2 ** 32 * pErr[1];
+      if (errPtr !== 0) {
+        const err = readCstr(errPtr);
+        sqlite3_free(errPtr);
+        throw new Error(err);
       }
-      stmt.step();
-      stmt.finalize();
-    } else sqlite3_exec(this.#handle, sqlStr);
-  }
+      return;
+    }
 
-  /**
-   * Creates a new prepared statement.
-   *
-   * Example:
-   * ```ts
-   * const stmt = db.prepare("insert into users (id, username) values (?, ?)");
-   *
-   * for (const user of usersToInsert) {
-   *   stmt.execute(id, user);
-   * }
-   *
-   * stmt.finalize();
-   * ```
-   *
-   * @param sql SQL string for prepared query.
-   * @returns A `PreparedStatement` object, on which you can call `execute` multiple
-   * times and then `finalize` it.
-   */
-  prepare(sql: string): PreparedStatement {
-    const handle = sqlite3_prepare_v2(this.#handle, sql);
-    return new PreparedStatement(this, handle);
+    const stmt = this.prepare(sql);
+    stmt.run(...params);
   }
 
-  /**
-   * Runs an SQL query with given parameters, and returns rows as array of columns.
-   * If you need the rows as objects, use `queryObject` instead. However, it is
-   * recommended to use `queryArray` because of the extra overhead added by FFI
-   * calls to get column names to create row objects.
-   *
-   * Example:
-   * ```ts
-   * const users = db.queryArray<[number, string]>("select id, username from users");
-   *
-   * // Using bind parameters
-   * const [user] = db.queryArray<[number, string]>("select id, username from users where email = ?", email);
-   *
-   * // Using template strings
-   * const [user] = db.queryArray<[number, string]>`select id, username from users where email = ${email}`;
-   *
-   * // Using named bind parameters
-   * const [user] = db.queryArray<[number, string]>("select id, username from users where email = :email", { email });
-   * ```
-   *
-   * @param sql SQL query to execute.
-   * @param args Parameters to bind to the query.
-   *
-   * @returns Array of rows (where rows are containing array of columns).
-   */
-  queryArray<T extends unknown[] = any[]>(
```

Comment:
should we add shorthands for querying like this?

Reply:
Hello - I know you're not asking me, but... As a user - I didn't care, as long as I can do whatever I need to with a prepared statement.

Reply:
thanks for the feedback, we can skip this one for now then